docs/DeveloperGuide.md

```diff
@@ -296,6 +296,151 @@ The following sequence diagram shows how the flow of the example execution:
+    T apply(String str) throws IllegalArgumentException, ParseException;
+}
+```
+The last parameter, `isOptional`, is a boolean flag that indicates whether the flag is optional or not. If the flag is not found in the user input, this method will throw a `ParseException` if the flag is not optional, but will return a `null` object if the flag is optional. It is overloaded to has a default value of `true` when omitted.
+
+
+The `parseField` method is reused extensively in multiple parser classes to parse all the flags. The following is an example of how it can be used:
+```
+// Parse the name flag, which is compulsory
+Name name = TypeParsingUtil.parseField("name", args, Name::of, false);
+
+// Parse the phone flag, which is optional
+Phone phone = TypeParsingUtil.parseField("phone", args, Phone::of);
+```
+##### 3. Parsing of the EditPersonCommand
+The `EditPersonCommandParser` class will parse the user input into an `EditPersonCommand` object. 
+
+This is achieved by combining the results of parsing flags in the previous stage and construct a `Person` object that contains the updated fields.
+
+Each of the `ListEntryField` has a global and static singleton default instance. For example, the default `Phone` is a `Phone` object with message "To be added."Please also note that it is not possible to create a `Phone` object with such message via public constructor and factory method.
+
+Each of the `ListEntry` also has a global and static singleton default instance whose fields are all the default `ListEntryField` objects. We can obtain a clone of the default `ListEntry` object via a static method in the class method.
+
+The `EditPersonCommandParser` class will first obtain a clone of the default `Person` object via the `getDefault` method, and then update the fields of the `Person` object with the results of parsing flags in the previous stage when the corresponding flags are found in the user input.
+
+The following is an example of how it can be used:
+```
+Person person = Person.getDefaultPerson();
+            person.setNameIfNotDefault(parseField("name", args, Name::of, nameIsOptional));
+            person.setPhoneIfNotDefault(parseField("phone", args, Phone::of));
+            person.setEmailIfNotDefault(parseField("email", args, Email::of));
+            person.setAddressIfNotDefault(parseField("address", args, Address::of));
+            person.setSubjectsIfNotDefault(parseField("subject", args, Subjects::of));
+            person.setTagsIfNotDefault(parseField("tag", args, Tags::of));
+            person.setRemarkIfNotDefault(parseField("remark", args, Remark::of));
+```
+##### 4. EditPersonCommand Implementation
+The `EditPersonCommand` class will execute the command and update the model accordingly.
+
+Realising the commonality between all kinds of edit commands, we have created an abstract generic class, `EditCommand`, to serve as template class containing the logic of editing a `ListEntry` object.
```

Comment:
should this be `AbstractEditCommand` instead of just `EditCommand`

Reply:
Thank you for pointing that out. Fixed in the latest commit.